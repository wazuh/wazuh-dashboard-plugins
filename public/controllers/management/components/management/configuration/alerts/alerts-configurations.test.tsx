import React from 'react';
import WzConfigurationAlerts from './alerts';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

jest.mock('../../../../../../kibana-services', () => ({
    getUiSettings:() => ({
        get:() => {
            return false
        }
    }),
}));

const mockProps = {
    "clusterNodeSelected":"master-node",
    "agent":{
       "id":"000"
    },
    "refreshTime":false,
    "currentConfig":{
       "analysis-alerts":{
          "alerts":{
             "email_alert_level":12,
             "log_alert_level":3
          }
       },
       "analysis-labels":{
          "labels":[
             
          ]
       },
       "mail-alerts":"Fetch configuration. 3013 - Error connecting with socket",
       "monitor-reports":{
          
       },
       "csyslog-csyslog":"Fetch configuration. 3013 - Error connecting with socket"
    },
    "wazuhNotReadyYet":""
 }


const mockStore = configureMockStore();
const store = mockStore({});

describe('WzConfigurationAlerts component mount OK', () => {

  it('renders correctly to match the snapshot', () => {
    const wrapper = shallow(
        <Provider store={store}>
            <WzConfigurationAlerts {...mockProps} /> 
        </Provider>
        );
    expect(wrapper).toMatchSnapshot();
  });

});