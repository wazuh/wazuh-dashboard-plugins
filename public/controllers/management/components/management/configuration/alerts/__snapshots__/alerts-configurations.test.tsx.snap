// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WzConfigurationAlerts component mount OK renders correctly to match the snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(WithLoading(Connect(WzConfigurationAlerts)))
    agent={
      Object {
        "id": "000",
      }
    }
    clusterNodeSelected="master-node"
    currentConfig={
      Object {
        "analysis-alerts": Object {
          "alerts": Object {
            "email_alert_level": 12,
            "log_alert_level": 3,
          },
        },
        "analysis-labels": Object {
          "labels": Array [],
        },
        "csyslog-csyslog": "Fetch configuration. 3013 - Error connecting with socket",
        "mail-alerts": "Fetch configuration. 3013 - Error connecting with socket",
        "monitor-reports": Object {},
      }
    }
    refreshTime={false}
    wazuhNotReadyYet=""
  />
</ContextProvider>
`;
